### PR TITLE
feat(resource-room): empty state for empty resource room

### PR DESCRIPTION
**Pull request by kishore03109**
## Problem
Currently, our empty state for ResourceRoom is not very pleasant to look at and not intuitive. Isomer should have a more descriptive empty state to guide the user along

Addresses #962

## Solution
As suggested in [figma here](https://www.figma.com/file/7OtNYkc4s9BkFk05ConEbd/V1.5?node-id=1559%3A62055)

**Breaking Changes**

- [x] No - this PR is backwards compatible

## Before & After Screenshots

**BEFORE**:

before

**AFTER**:

Screenshot 2022-07-26 at 12 24 38 PM

Other Notes: 
Other changes are linked to outstanding PRs by @seaerchin

### CODE REVIEW

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (kishore03109) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:133]: @seaerchin
The eslint gives and error saying: "Prop spreading is forbidden"

However, I do see this in other areas of the codebase.. Are we ignoring this warning as a team for now?

**Comment by seaerchin** [comment]
@kishore03109 I changed the base branch for you; could you rebase to remove the extra commits?

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [reply src/layouts/ResourceRoom/ResourceRoom.tsx:133]: Hmm, this should be a warning and not an error; I'm in favour of ignoring this as this is a design pattern given by the `react-hook-form` docs and we shouldn't need to care about the individual props returned

**Comment by kishore03109** [comment]
@seaerchin I was about to commit to remove the changes for the changelog and package.json... but would it actually be easier for you when we merge this into master? do you want me to remove the or nah?

**Comment by seaerchin** [comment]
> @seaerchin I was about to commit to remove the changes for the changelog and package.json... but would it actually be easier for you when we merge this into master? do you want me to remove the or nah?

We should remove! The merging strategy here is for all of these commits to go into the tracking branch for the page level stories (#980), which would then be merged back into `develop`. 

As `develop` already contains these commits (and the commits are non-overlapping), keeping the changes to `package.json`/`changelog.md` is unnecessary and might wonk up the merge later ._. 

Also, we **don't** merge into `master` normally (this is only during release) and instead, merge into `develop`

File: src/layouts/ResourceCategory/ResourceCategory.tsx
Comment (seaerchin) [inline src/layouts/ResourceCategory/ResourceCategory.tsx:51]: supernit: `arePagesEmpty` or `hasPages` (but needs conditional change)

File: src/layouts/ResourceCategory/ResourceCategory.tsx
Comment (seaerchin) [inline src/layouts/ResourceCategory/ResourceCategory.tsx:75]: should we account for the fact that `isPagesEmpty` might change state due to loading? a probable sequence of events might be `true` (as before first fetch it defaults to `undefined`) -> successful load -> `false`, leading to this to be shown.

a quick fix would just be to have a skeleton wrap this (or to subsume it into the skeleton below)

File: src/layouts/ResourceCategory/ResourceCategory.tsx
Comment (seaerchin) [inline src/layouts/ResourceCategory/ResourceCategory.tsx:90]: this is kinda weird - the `Center` already guarantees that both are vertically/horizontally centered; we should instead structure this like so:

```ts
<Center>
  <Text textStyle = "subhead-1">
    There&apos;s nothing here yet.
  </Text>
  <Text textStyle = "body-2">
    Create a resource page to get started.
  </Text>
</Center>
```

which prevents ambiguity. wdyt about this change?

File: src/layouts/ResourceCategory/ResourceCategory.tsx
Comment (seaerchin) [inline src/layouts/ResourceCategory/ResourceCategory.tsx:84]: if we combine the `Center`, we shouldn't need this `VStack` anymore

File: src/layouts/ResourceRoom/ResourceRoom.stories.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.stories.tsx:69]: could we shift this into the decorator above? it'd still apply to this and it'd be neater

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:109]: see prior comment about combining `Center` and using `Text`

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:98]: this is slightly weird - why do we need `mt = {100}`

File: src/layouts/ResourceCategory/ResourceCategory.tsx
Comment (seaerchin) [inline src/layouts/ResourceCategory/ResourceCategory.tsx:93]: idt we need this prop as the default variant is already `solid`

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:111]: extra prop

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:120]: could i check why scrolling is blocked on mount? if this is just to maintain compat w/ current behaviour, we should double check with the designers on the _intended_ behaviour

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:147]: supernit: could we change `colorScheme` to `textColor` as chakra actually leaves some `colorSchemes` unimplemented and it might be confusing in the future.

also, [this prop](https://chakra-ui.com/docs/components/text/props) doesn't seem to be implemented so i'm not q sure if this is working as expected

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:155]: oops, could we use `Button` and pass the `isLoading` prop? this component is legacy (should've marked it in JSDoc as dep also psps) and it's not ideal to use it.

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:146]: could we double check if this is supposed to be `link` or `clear`?

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:246]: we can inline this below unless the concern is the `Skeleton`

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:270]: see above wrt combining

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:272]: extra prop

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:153]: does this work? there doesn't seem to be a `form` higher up in the DOM from what i can tell

File: src/assets/images/EmptyBoxImage.tsx
Comment (seaerchin) [inline src/assets/images/EmptyBoxImage.tsx:2]: should we spread this over the `svg`?

**Review by seaerchin — CHANGES_REQUESTED** [review]

**Comment by seaerchin** [comment]
need rebase, feel free to ping me on slack when done

File: src/assets/images/EmptyBoxImage.tsx
Comment (seaerchin) [inline src/assets/images/EmptyBoxImage.tsx:1]: we should have a props here that we spread over the `svg` - this allows consumers to style the `svg` if needed

File: src/layouts/ResourceCategory/ResourceCategory.tsx
Comment (seaerchin) [inline src/layouts/ResourceCategory/ResourceCategory.tsx:65]: nit: evaluate

File: src/layouts/ResourceRoom/ResourceRoom.stories.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.stories.tsx:70]: just an fyi, i actually added these stories upstream also - if your PR gets merged into `develop` first, i'll remove mine!

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:99]: should we prefer using fragments for this use case? ie, 
```ts
<>
<SiteViewLayout />
<Modal />
</>
```

this avoids having to sidestep the divider

**Review by seaerchin — CHANGES_REQUESTED** [review]

File: src/components/EmptyArea.tsx
Comment (seaerchin) [inline src/components/EmptyArea.tsx:6]: nit: could we rename this to `EmptyAreaProps`? i think the convention is to use `xxxProps` so we should stick to it!

also, we should probably extend `StackProps` from chakra as the wrapper element is a `VStack`

File: src/components/EmptyArea.tsx
Comment (seaerchin) [inline src/components/EmptyArea.tsx:8]: this might be wrong - `ReactFragment` specifically refers to `<> </>` ._.  as an aside, this actually accepts jsx elements, which i didn't know so til! 

the side effect of this is that now, this becomes a required prop, which isn't ideal - wdyt about chagning this to `JSX.Element`?

File: src/components/EmptyArea.tsx
Comment (seaerchin) [inline src/components/EmptyArea.tsx:7]: this should be an optional prop - this makes the API stricter without offering additional benefits. eg: if an initial call to an API returns `undefined`, the user now has to do a cast, which is additional friction for this component

File: src/components/EmptyArea.tsx
Comment (seaerchin) [inline src/components/EmptyArea.tsx:1]: there's nothing wrong with putting this in `src/components` but this implies that the # of use-cases this component has to cover would be much wider than if it was limited. 

this impacts how the component should be structured - eg: presence of button/text. one way people get around this (see: chakra's [menu](https://chakra-ui.com/docs/components/menu#menubutton-props) or [modal](https://chakra-ui.com/docs/components/modal#block-scrolling-when-modal-opens)) is by having many related components grouped together and exported under a common prefix (or as per design system, under a grouped export).

the rationale to allow for more flexibility here is good (by adopting optional arguments) - but this makes a judgement that the text **has to exist**, which might not be ideal. could i check here if this has been cleared by our designers and also, that empty states going forward have been tied down to this

File: src/components/EmptyArea.tsx
Comment (seaerchin) [inline src/components/EmptyArea.tsx:23]: why do we need a `form` here?

File: src/components/EmptyArea.tsx
Comment (seaerchin) [inline src/components/EmptyArea.tsx:19]: we've given this element a more powerful api by extending chakra props but the extra flexibility isn't being used - should we consider having a `...rest` argument which is spread over the `VStack`?

File: src/layouts/ResourceRoom/ResourceRoom.stories.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.stories.tsx:63]: supernit: could we change this to `EmptyResourceCategory`

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.tsx:365]: can just ignore cos it's a warning hahahahahah

File: src/layouts/layouts/SiteViewLayout/SiteViewLayout.tsx
Comment (seaerchin) [inline src/layouts/layouts/SiteViewLayout/SiteViewLayout.tsx:33]: can remove if u wanna but it's fine to leave too

File: src/layouts/ResourceRoom/ResourceRoom.stories.tsx
Comment (seaerchin) [inline src/layouts/ResourceRoom/ResourceRoom.stories.tsx:64]: nit: rename

File: src/components/EmptyArea.tsx
Comment (seaerchin) [inline src/components/EmptyArea.tsx:6]: could we make this prop optional?

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [reply src/layouts/ResourceRoom/ResourceRoom.tsx:104]: i think this^ looks better instead of having `divider = {undefined}` - wdyt?

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [reply src/layouts/ResourceRoom/ResourceRoom.tsx:155]: could we do this^

File: src/components/EmptyArea.tsx
Comment (seaerchin) [reply src/components/EmptyArea.tsx:6]: If they don't pass it, won't it default to `undefined` and hence, we can still do a type conversion under the hood?

File: src/components/EmptyArea.tsx
Comment (seaerchin) [reply src/components/EmptyArea.tsx:1]: Sure! I think this is just something to be aware of next time. Wrt this component, it's general enough to be placed here and the given API is fine (if text is optional).

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [reply src/layouts/ResourceRoom/ResourceRoom.tsx:155]: Actually this is because the component has crashed before due to the `useEffect` ._.

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [reply src/layouts/ResourceRoom/ResourceRoom.tsx:148]: I think we should be using design system's `Button`, and the equivalent variant is `clear`; sorry if this was unclear (pun intended)

File: src/components/EmptyArea.tsx
Comment (kishore03109) [reply src/components/EmptyArea.tsx:6]: Hmm when would you have a case where I wont pass it in `isItemEmpty` first place ah? This JSX element is designed such that `isItemEmpty` is `true`, it would show the empty state, else it would render its `children`. I cant think of any case that we would need only show the empty state at **all** times, am I mistaken here?

File: src/components/EmptyArea.tsx
Comment (kishore03109) [reply src/components/EmptyArea.tsx:6]: to be discussed under here [instead](https://github.com/isomerpages/isomercms-frontend/pull/993#discussion_r959316420)

File: src/layouts/ResourceRoom/ResourceRoom.stories.tsx
Comment (kishore03109) [reply src/layouts/ResourceRoom/ResourceRoom.stories.tsx]: ee2d6e

File: src/components/EmptyArea.tsx
Comment (kishore03109) [reply src/components/EmptyArea.tsx:6]: (repost because Idk why the github is duplicating shit ._.)
hmm I disagree here: isItemEmpty is used [here](https://github.com/isomerpages/isomercms-frontend/blob/add-empty-state-for-empty-resource/src/components/EmptyArea.tsx#L22) If I were to make this optional, then developers might not pass this parameter when calling EmptyArea, which then renders this whole component useless.

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (kishore03109) [reply src/layouts/ResourceRoom/ResourceRoom.tsx:148]: +1 for comment -1 for lame pun

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (kishore03109) [reply src/layouts/ResourceRoom/ResourceRoom.tsx:104]: eh I think got misunderstanding here bah - this components  for this empty state will be shown below the entire page leh, unless we redefine what `SiteViewLayout` does which I dont feel should be the case here

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (kishore03109) [reply src/layouts/ResourceRoom/ResourceRoom.tsx:148]: 03555cc

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [reply src/layouts/ResourceRoom/ResourceRoom.tsx:105]: hmm i can't repro this! feel free to ping me to pair on it

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [reply src/layouts/ResourceRoom/ResourceRoom.tsx:154]: hmm from the [spec](https://www.w3.org/TR/html4/interact/forms.html#buttons) itself, they say that 
> controls outside a form cannot be [successful controls](https://www.w3.org/TR/html4/interact/forms.html#successful-controls).

which seems to imply that a `form` element is required .-.

**Review by seaerchin — COMMENTED** [review]
overall looks fine! just 2 minor comments wrt dom structure. will approve once addressed!

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [reply src/layouts/ResourceRoom/ResourceRoom.tsx]: this actually seems fine now!

**Comment by kishore03109** [comment]
> not about the errors but more of setting the type = "submit", which requires that there's a DOM node higher up. if you've verified that this works, i don't have an issue with it (apart from figuring out how it works AHHA)

@seaerchin correct me if I am wrong ya, but the reason this works is because the `handleSubmit` is from the `useForm` hook (This still works when I completely remove the `type` attribute as a whole)

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [reply src/layouts/ResourceRoom/ResourceRoom.tsx]: resolved offline - think the `onClick` is the one that allows this to work

File: src/layouts/ResourceRoom/ResourceRoom.tsx
Comment (seaerchin) [reply src/layouts/ResourceRoom/ResourceRoom.tsx]: fixed

File: src/components/EmptyArea.tsx
Comment (seaerchin) [inline src/components/EmptyArea.tsx:5]: nit: change htis to `StackProps`

**Review by seaerchin — APPROVED** [review]
lgtm with minor nit

**Pull request merged**
